src/vcpkg-test/dependencies.cpp

```diff
@@ -215,6 +216,25 @@ struct MockOverlayProvider : IOverlayProvider
 
 static const MockOverlayProvider s_empty_mock_overlay;
 
+static void CHECK_LINES(const LocalizedString& a, StringView b)
+{
+    auto as = Strings::split(a.data(), '\n');
+    auto bs = Strings::split(b, '\n');
+    for (size_t i = 0; i < as.size() && i < bs.size(); ++i)
+    {
+        INFO(i);
+        CHECK(as[i] == bs[i]);
+    }
+    CHECK(as.size() == bs.size());
+}
+
+template<class T>
+static T require(ExpectedL<T>&& e)
```

Comment:
This has the same effect as calling `value_or_exit` so I think callers should do that.

Reply:
The difference is that value_or_exit results in failfast with no test information -- this function integrates correctly with Catch2, providing the test error stack.

Reply:
It definitely does *not* give you a stack -- that's my objection to the CHECK_LINES above. catch will complain about the location where its macro is, in this case the require, which loses all context about the failure.

At least `value_or_exit` prints the correct test line where there is a problem rather than making every failure print the same line (inside `require`)

Reply:
Turning it into a macro solves the line info issue.